### PR TITLE
Fix multiline formatting bug

### LINKED ISSUE #2: formatting of multi-line annotation text only applies to first line (answer: now working)
Is there any way of getting e.g. `\sffamily` to apply across multiple lines?

### PR DESCRIPTION
**Pull request by lenianiva**
Resolves #2

### CODE REVIEW

**Comment by lenianiva** [comment]
This branch replaces `\eqnannotationtext` with `\eqnannotationfont` and `\eqnannotationstrut` with the font part provided to `\node[font=\eqnannotationfont]`. I have not completely modified the example and PDF yet in order to showcase that it works as intended.

**Comment by st--** [comment]
Great, thank you so much for contributing - I'll see how it works :)

**Comment by st--** [comment]
Just to let you know I'm still intending to sort this out!

**Comment by lenianiva** [comment]
Thanks! (sorry i replied to the wrong thread that is why you see the edit)

File: annotate-equations.sty [hunk 1]
```diff
@@ -126,10 +127,11 @@
     \def\myEAcolor{\usevalue{\myEAmarkOne}}%
     \begin{tikzpicture}[overlay,remember picture,>=stealth,nodes={align=left,inner ysep=1pt},<-]
         % default anchor is at center
-        \node[anchor=\swapNorthSouth{\EAmarkanchor},color=\myEAcolor!85,#1]  % color blended with white to 85%, any (optional) extra args #1
+		\node[anchor=\swapNorthSouth{\EAmarkanchor},color=\myEAcolor!85,#1,
+			font=\eqnannotationfont]  % color blended with white to 85%, any (optional) extra args #1
```
Comment (st--) [inline annotate-equations.sty]:
```suggestion
		\node[anchor=\swapNorthSouth{\EAmarkanchor},color=\myEAcolor!85,font=\eqnannotationfont,
			#1]  % color blended with white to 85%, any (optional) extra args #1
```
the `#1` should be last so the user can override previously set options

File: annotate-equations.tex [hunk 2]
```diff
@@ -217,10 +217,11 @@ \subsection{Amount of shading of mark highlight}
 
 \subsection{Default formatting of annotation labels}
 
-\verb|\eqnannotationtext| is a one-argument command that gets the annotation text as an argument and can be used to have consistent formatting:
+\verb|\eqnannotationfont| is a one-argument command that feeds into the
+\texttt{font} field of tikz and can be used to have consistent formatting:
```
Comment (st--) [inline annotate-equations.tex]:
```suggestion
\texttt{font} field of TikZ and can be used to have consistent formatting:
```

File: annotate-equations.tex [hunk 3]
```diff
@@ -242,7 +241,32 @@ \subsection{Default formatting of annotation labels}
 \vspace{1em}
 \end{LTXexample}
 \noindent
+
+\verb|\eqnannotationstrut| is defined to be a strut to provide minimum distance
+from text to the surrounding diagrams. By default it is \verb|\strut|, which has
```
Comment (st--) [inline annotate-equations.tex]:
```suggestion
\verb|\eqnannotationstrut| is defined to be a strut (zero-width height) to provide minimum distance between the text and the corresponding arrow line. By default it is \verb|\strut|, which has
```

File: annotate-equations.tex [hunk 4]
```diff
@@ -242,7 +241,32 @@ \subsection{Default formatting of annotation labels}
...
+a similar effect to \verb|\mathstrut| in \verb|\eqnhighlightheight|.
+
+\begin{LTXexample}[text outside listing,lefthand width=0.5in]
+% Default
+\renewcommand{\eqnannotationstrut}{\strut}
+\begin{equation*}
+	\eqnmarkbox[blue]{v}{v}
+\end{equation*}
+\annotate[yshift=-0.5em]{below}{v}{velocity}
```
Comment (st--) [inline annotate-equations.tex]:
```suggestion
    \eqnmarkbox[blue]{size}{s}
\end{equation*}
\annotate[yshift=-0.5em]{below}{size}{the size}
```
I think the effect is more striking with text that doesn't have descenders (like in `y`).
also, changed the tab to spaces for consistency with the rest of the formatting.

File: annotate-equations.tex [hunk 5]
```diff
@@ -242,7 +241,32 @@ \subsection{Default formatting of annotation labels}
...
 \end{LTXexample}
 \noindent
+
+\verb|\eqnannotationstrut| is defined to be a strut to provide minimum distance
+from text to the surrounding diagrams. By default it is \verb|\strut|, which has
+a similar effect to \verb|\mathstrut| in \verb|\eqnhighlightheight|.
+
+\begin{LTXexample}[text outside listing,lefthand width=0.5in]
+% Default
+\renewcommand{\eqnannotationstrut}{\strut}
```
Comment (st--) [inline annotate-equations.tex]:
```suggestion
\renewcommand{\eqnannotationstrut}{\strut}  % package default
```
for consistency

File: annotate-equations.tex [hunk 6]
```diff
@@ -242,7 +241,32 @@ \subsection{Default formatting of annotation labels}
...
+\noindent
 %
+\begin{LTXexample}[text outside listing,lefthand width=0.5in]
+	\renewcommand{\eqnannotationstrut}{}
+
+	\begin{equation*}
+    \eqnmarkbox[blue]{v}{v}
+	\end{equation*}
+	\annotate[yshift=-0.5em]{below}{v}{velocity}
+	\vspace{1em}
```
Comment (st--) [inline annotate-equations.tex]:
```suggestion
\renewcommand{\eqnannotationstrut}{}
\begin{equation*}
    \eqnmarkbox[blue]{size}{s}
\end{equation*}
\annotate[yshift=-0.5em]{below}{size}{the size}
\vspace{1em}
```
in line with above

File: annotate-equations.tex [hunk 7]
```diff
@@ -242,7 +241,32 @@ \subsection{Default formatting of annotation labels}
...
+\end{LTXexample}
+\noindent
```
Comment (st--) [inline annotate-equations.tex]: I don't think this is necessary here (at the end of a section)?
```suggestion
\noindent
```

File: annotate-equations.tex [hunk 8]
```diff
@@ -311,7 +335,6 @@ \section{Known issues}
 \begin{itemize}
     \item \texttt{label above}/\texttt{label below} is not implemented for \verb|\annotate|.
 
-    \item Formatting only covers first line in multi-line annotation texts (see \cref{sec:multiline}).
```
Comment (st--) [inline annotate-equations.tex:314]: Thanks for fixing it :tada:

File: annotate-equations.tex [hunk 9]
```diff
@@ -263,7 +287,7 @@ \subsection{Line breaks within annotations}
 \end{LTXexample}
 \noindent
 %
-Here is a manual work-around:
+Here is a manual work-around: (obsolete)
```
Comment (st--) [inline annotate-equations.tex]: I don't think it's helpful to show the old work-around - could you just delete it instead? All it needs to say here is that using `\\` within the text to get a line break works just fine.:)

**Review by st-- — CHANGES_REQUESTED** [review]
Hi @vleni ! Thank you so much for fixing this issue. This is basically good to go, I just marked a few changes, mainly for consistency - most of it as review suggestions, so you could just go and accept them from GitHub. The only other thing I couldn't "suggest" directly was removing the now-obsolete workaround - I've left a comment about that.

If you're happy with my suggestions, ping me and I'll merge it in. :)

**Comment by lenianiva** [comment]
Thanks! It should be ok now after your change.

**Comment by st--** [comment]
@vleni if you agree with all the suggestions, you can just go and accept them (see [GitHub docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) in case that's new to you).

Also, would you be happy to adapt the no-longer-needed section on the workaround (see https://github.com/st--/annotate-equations/pull/11/files#r1105454093) instead of just marking it "obsolete"?

If you prefer, I can also do all those changes myself, but I didn't want to do so without giving you the chance first because it would seem rude to me to edit someone else's contributions for them:)

**Comment by st--** [comment]
(Side note- I edited the PR description to include "resolves #2" because a) the link to the issue makes it easier to keep up with why this PR is here and what it does and b) GitHub has [this neat feature](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) where merging a PR also closes the linked issues :))

**Comment by st--** [comment]
Oh, one more task: after the final changes, could you also regenerate the PDF & commit that?

_Update:_ I spotted a minor typography issue in the TeX that I fixed on the `main` branch - you can either merge that into your branch and then regenerate the PDF, or skip generating the PDF & I'll do it separately after merging your PR - either works:)

**Comment by lenianiva** [comment]
ok I fixed all of the suggestions and I regenerated the PDF but I didn't merge master. Could you do it in the merge commit?

File: annotate-equations.tex [hunk 10]
```diff
@@ -217,10 +217,11 @@ \subsection{Amount of shading of mark highlight}
 
 \subsection{Default formatting of annotation labels}
 
-\verb|\eqnannotationtext| is a one-argument command that gets the annotation text as an argument and can be used to have consistent formatting:
+\verb|\eqnannotationfont| is a one-argument command that feeds into the
+\texttt{font} field of TikZ and can be used to have consistent formatting:
```
Comment (st--) [inline annotate-equations.tex]:
```suggestion
\verb|\eqnannotationfont| sets the \texttt{font} field of the TikZ annotation label and can be re-set to change its formatting:
```

File: annotate-equations.tex [hunk 11]
```diff
@@ -242,37 +241,38 @@ \subsection{Default formatting of annotation labels}
 \vspace{1em}
 \end{LTXexample}
 \noindent
-%
 
+\verb|\eqnannotationstrut| is defined to be a strut (zero-width height) to
+provide minimum distance between the text and the corresponding arrow line. By
+default it is \verb|\strut|, which has, which has a similar effect to
```
Comment (st--) [inline annotate-equations.tex]:
```suggestion
default it is \verb|\strut|, which has a similar effect to
```

**Review by st-- — COMMENTED** [review]
looks good!

**Pull request merged**